src/main/java/core/basesyntax/io/FileWriterImpl.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.io;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+
+public class FileWriterImpl implements FileWriter {
+    @Override
+    public void write(String reportGenerator, File output) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new java.io.FileWriter(output))) {
+            bufferedWriter.write(reportGenerator);
```

No validation of input parameters is performed. Add checks such as `if (content == null) { throw new RuntimeException("Report content cannot be null"); }` and `if (output == null)` (or for the new API `filePath`), and fail fast with informative messages. This ensures calling code gets clear errors instead of NPEs or IO exceptions later.